src/calendar.cpp

```diff
@@ -556,11 +556,24 @@ time_duration rng( time_duration lo, time_duration hi )
 
 season_type season_of_year( const time_point &p )
 {
-    if( calendar::eternal_season() ) {
-        // If we use calendar::start to determine the initial season, and the user shortens the season length
-        // mid-game, the result could be the wrong season!
-        return calendar::initial_season;
+    static time_point prev_turn = calendar::before_time_starts;
+    static season_type prev_season = SPRING;
```

Comment:
may as well initialize this as `calendar::initial_season`

Reply:
Yeah, you're right, might as well.